lib/myhttp.js

```diff
@@ -117,4 +94,45 @@ Client.prototype.destroy = function () {
   this.conn.destroy()
 }
 
+Client.prototype.setHeaders = function (newHeaders) {
+  this.opts.headers = newHeaders || {}
+  this.rebuildRequest()
+}
+
+Client.prototype.setBody = function (newBody) {
+  this.opts.body = newBody
+  this.body = newBody
+
+  this.rebuildRequest()
+}
+
+Client.prototype.rebuildRequest = function () {
+  // this.body should take precedence, as the user can reassign directly to it
+  this.opts.body = this.body || this.opts.body
+
+  let bodyReq
+
+  if (typeof this.opts.body === 'string') {
+    bodyReq = new Buffer(this.opts.body)
+  } else if (Buffer.isBuffer(this.opts.body)) {
+    bodyReq = this.opts.body
+  } else if (this.opts.body) {
+    throw new Error('body must be either a string or a buffer')
+  }
+
+  if (bodyReq) {
+    this.opts.headers['Content-Length'] = '' + bodyReq.length
+  }
+
+  this._req = Object.keys(this.opts.headers)
+    .map((key) => `${key}: ${this.opts.headers[key]}\r\n`)
+    .reduce((acc, str) => acc + str, this._req)
```

Comment:
Well, the request is built/cached once, unless users go crazy with the `.setHeaders()` and `.setBody()` call. If you this improved though, I'll need some guidance :)